internal/test/framework/framework.go

```diff
@@ -0,0 +1,198 @@
+	instanceType := "m4.large"
+	sshKey := "libra"
+	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, awsCredentials, "default", artifactDir,
+		imageID, instanceType, sshKey, privateKeyPath)
+	if err != nil {
+		return fmt.Errorf("error instantiating cloud provider %v", err)
+	}
+	f.cloudProvider = cloud
+	credentials, err := cloud.CreateWindowsVM()
+	if err != nil {
+		return fmt.Errorf("error creating Windows VM: %v", err)
+	}
+	f.Credentials = credentials
+	return nil
+}
+
+// setupWinRMClient sets up the winrm client to be used while accessing Windows node
+func (f *TestFramework) setupWinRMClient() error {
+	host := f.Credentials.GetIPAddress()
+	password := f.Credentials.GetPassword()
+
+	endpoint := winrm.NewEndpoint(host, winRMPort, true, true, nil, nil, nil, 0)
+	winrmClient, err := winrm.NewClient(endpoint, user, password)
+	if err != nil {
+		return fmt.Errorf("failed to set up winrm client with error: %v", err)
+	}
+	f.WinrmClient = winrmClient
+	return nil
+}
+
+// configureOpenSSHServer configures the OpenSSH server using WinRM client installed on the Windows VM.
+// The OpenSSH server is installed as part of WNI tool's CreateVM method.
+func (f *TestFramework) configureOpenSSHServer() error {
+	// This dependency is needed for the subsequent module installation we're doing. This version of NuGet
+	// needed for OpenSSH server 0.0.1
+	installDependentPackages := "Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force"
+	if _, err := f.WinrmClient.Run(remotePowerShellCmdPrefix+installDependentPackages,
+		os.Stdout, os.Stderr); err != nil {
+		return fmt.Errorf("failed to install dependent packages for OpenSSH server with error %v", err)
+	}
+	// Configure OpenSSH for all users.
+	// TODO: Limit this to Administrator.
+	if _, err := f.WinrmClient.Run(remotePowerShellCmdPrefix+"Install-Module -Force OpenSSHUtils -Scope AllUsers",
+		os.Stdout, os.Stderr); err != nil {
+		return fmt.Errorf("failed to configure OpenSSHUtils for all users with error %v", err)
+	}
+	// Setup ssh-agent Windows Service.
+	if _, err := f.WinrmClient.Run(remotePowerShellCmdPrefix+"Set-Service -Name ssh-agent -StartupType ‘Automatic’",
+		os.Stdout, os.Stderr); err != nil {
+		return fmt.Errorf("failed to set up ssh-agent Windows Service with err %v", err)
+	}
+	// Setup sshd Windows service
+	if _, err := f.WinrmClient.Run(remotePowerShellCmdPrefix+"Set-Service -Name sshd -StartupType ‘Automatic’",
+		os.Stdout, os.Stderr); err != nil {
+		return fmt.Errorf("failed to set up sshd Windows Service with err %v", err)
+	}
+	if _, err := f.WinrmClient.Run(remotePowerShellCmdPrefix+"Start-Service ssh-agent",
+		os.Stdout, os.Stderr); err != nil {
+		return fmt.Errorf("start ssh-agent with error %v", err)
+	}
+	if _, err := f.WinrmClient.Run(remotePowerShellCmdPrefix+"Start-Service sshd",
+		os.Stdout, os.Stderr); err != nil {
+		return fmt.Errorf("failed to start sshd with error %v", err)
+	}
+	return nil
```

Comment:
In the above message, please replace `with error` with `:` to make the message more succinct.

Reply:
Done